fix(dashboard): Update filter logic to use nodes as source [#294]

LGSI-KarumuriHari opened the pull request:
📝 PR Description
This PR refactors the dashboard workloads logic to remove all mock node data and implements real-time node and pod updates. Nodes are now fetched from the backend API every 5 seconds, ensuring the dashboard reflects the latest cluster state. Pod age and memory usage calculations have been improved for accuracy. The filtering logic has been updated to use nodes as the primary source, mapping pods to nodes after node creation. This change ensures that pods are always associated with the correct, up-to-date node information.

Key Changes
Removed all mock node data from the workloads , dashboard .
Added periodic fetch (every 5 seconds) for nodes from the backend API.
Improved pod age and memory usage calculation logic.
Updated filter logic: nodes are created first, then pods are mapped to nodes.
Enhanced UI to reflect real-time node and pod state.
Cleaned up unused code and comments for clarity.

## ScreenShots

Before: Even the node info is available , the filter is expecting from pods/containers data to map the nodes 
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/203277b7-f468-48a3-8436-d94ac8b63947" />

After : Changed the logic to fetch from nodes data to map for filter component , removing dependency from pods , and removed mock data if no pods are available 
<img width="1238" height="733" alt="image" src="https://github.com/user-attachments/assets/6e7251e0-e1b7-45b3-9704-33f66cb6eb6e" />

step1 : when only node data is present

<img width="1920" height="1080" alt="Screenshot from 2025-09-29 20-08-17" src="https://github.com/user-attachments/assets/1313893e-0eee-49ad-adbc-653bc901681c" />

step2: when we run .helloworld.sh , which is dynamically updated , as the fetch logic is being called every 5 secs for new updates
<img width="1920" height="1080" alt="Screenshot from 2025-09-29 20-09-30" src="https://github.com/user-attachments/assets/dae45c0a-aaf1-46ea-94e6-43342e092f08" />

🔗 Related Issue
closes->https://github.com/eclipse-pullpiri/pullpiri/issues/294

🧪 Test Method
logs are printed in console on successful fetch

✅ Checklist
[✅] Code conventions are followed
[✅] Tests are added/modified
[✅] Documentation is updated (if necessary)

chaehee-lim commented on src/tools/dashboard/src/components/Workloads.tsx:
```diff
@@ -52,24 +52,38 @@ export function Workloads({ onPodClick, pods, setPods }: WorkloadsProps) {
-        console.error("❌ Nodes API fetch failed:", e);
-      });
+          console.error("❌ Nodes API fetch failed:", e);
+        });
```
@LGSI-KarumuriHari 
Could you please display the error log in the Recent events?

chaehee-lim commented on src/tools/dashboard/src/components/Dashboard.tsx:
```diff
@@ -89,78 +130,85 @@ export function Dashboard() {
+
+    fetchPods();
+    const interval = setInterval(fetchPods, 5000); // Refresh every 5 seconds
+    return () => clearInterval(interval);
```
@daeyoung-jeong-lge 
This seems like a good idea.
I'll take note of this and apply it later when I create the UI in the dashboard.

LGSI-KarumuriHari commented on src/tools/dashboard/src/components/Dashboard.tsx:
```diff
@@ -89,78 +130,85 @@ export function Dashboard() {
         setPodsFetchSuccess(false);
         console.error("❌ Pods API (metrics) fetch failed:", e);
       });
```
@chaehee-lim  Is this action required only for the failure cases? As , the fetch would happens every 5 secs or user specified timeout Success logs might flood the recent events and might miss out important information .

LGSI-KarumuriHari commented:
- Added Accordingly for recentLogs component from both Pods and Nodes info
<img width="1638" height="858" alt="Screenshot from 2025-09-30 12-55-57" src="https://github.com/user-attachments/assets/35d0211b-bc34-4395-b286-0da817bbf71a" />

- Added default Timeout in .env ask requested
